MOxUnit/moxunit_runtests.m

```diff
@@ -67,25 +71,40 @@
     end
 
     % initialize test results
-    test_init=MOxUnitTestReport(verbosity, fid);
+    suite_name=class(suite);
+    test_report=MOxUnitTestReport(verbosity, fid, suite_name);
 
     % run all tests
-    test_result=run(suite, test_init);
+    test_report=run(suite, test_report);
 
     % show summary of test result
-    disp(test_result);
+    disp(test_report);
+
+    % if xml output was requested, store it in a file
+    if ~isempty(junit_xml)
+        write_junit_xml(junit_xml, test_report);
+    end
 
     % return true if no errors or failures
-    result=wasSuccessful(test_result);
+    result=wasSuccessful(test_report);
+
 
+function write_junit_xml(fn, test_report)
+    fid=fopen(fn,'w');
+    file_closer=onCleanup(@()fclose(fid));
 
-function [verbosity,filenames,fid]=get_params(varargin)
+    xml=getSummaryStr(test_report,'xml');
+    fprintf(fid,'%s',xml);
```

Comment:
Needs to print
```
<?xml version="1.0" encoding="utf-8"?>
```
beforehand.

And preferably wrap the testsuite xml in a `<testsuites>...</testsuites>` tag.